### PR TITLE
Release 1.4.1

### PR DESCRIPTION
**Pull request by kingdonb**
I think this might need to be a new MINOR version based on some dependency updates that are also MINOR, but as I had intended for this to be a CVE-patching only release, I had initially opened it up as a PATCH release.

Before I try to fix any of that, I'd like to open this one up and let it run for e2e just to be sure if other changes are needed before this can be merged, as a MINOR or PATCH release, or whatever it turns out to be.

### CODE REVIEW

**Comment by kingdonb** [comment]
There is `docker.io/kingdonb/helm-operator:release-141-f3a9c907` which has built successfully on my local machine, I'm running Snyk to assess any CVE reports or vulnerabilities present in the base image.

It has a much better Snyk score outcome than if you compare against the last release:

```
$ docker scan docker.io/kingdonb/helm-operator:release-141-f3a9c907

Testing docker.io/kingdonb/helm-operator:release-141-f3a9c907...

Organization:      kingdonb
Package manager:   apk
Project name:      docker-image|docker.io/kingdonb/helm-operator
Docker image:      docker.io/kingdonb/helm-operator:release-141-f3a9c907
Platform:          linux/amd64
Base image:        alpine:3.13.7
Licenses:          enabled

✓ Tested 31 dependencies for known issues, no vulnerable paths found.

According to our scan, you are currently using the most secure version of the selected base image
```

I've also upgraded several dependencies including Helm, to address CVE reports that we received through Dependabot. (Thanks GitHub)

File: chart/helm-operator/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
-appVersion: "1.4.0"
-version: 1.2.0
```
Comment (kingdonb) [inline chart/helm-operator/Chart.yaml:3]: Reviewing the entire diff, this change jumped out at me as probably not correct, but after reviewing the diff history and 665d493f9ee65a74fe759ecdf94d4fde69ff7dad I can see that it is correct. The weird look here is an artifact of there being no housekeeping merges in the helm-operator `releasing.md` in the process after the last chart was published.

(I will release another chart version here for this one, `1.4.1`, same chart version as well. No separate PR is required, if I read the releasing protocol correctly. I will reread it before closing this review...)

**Comment by kingdonb** [comment]
Pushing another force commit here to update CHANGELOG for publication. I think this is ready, but still have to see an E2E pass.

**Review by hiddeco — CHANGES_REQUESTED** [review]
The version in https://github.com/fluxcd/helm-operator/blob/cb2962c1ad5d3327d46925b2312c5c4ff45cc304/docker/helm3.version needs to be bumped as well.

**Comment by kingdonb** [comment]
I don't see what failed in https://app.circleci.com/pipelines/github/fluxcd/helm-operator/1613/workflows/c7353826-1ce0-45c2-9e84-77dd93635c5e/jobs/1839/parallel-runs/0/steps/0-110

Just going to let it run again and see what happens this time 👍

**Review by kingdonb — COMMENTED** [review]
CI is passing now, 👍 LGTM

**Review by hiddeco — APPROVED** [review]
Patches LGTM, thank you @kingdonb :bow:

**Pull request merged**